core/src/main/java/org/apache/spark/shuffle/unsafe/UnsafeShuffleExternalSorter.java

```diff
@@ -86,6 +86,9 @@
 
   private final LinkedList<SpillInfo> spills = new LinkedList<SpillInfo>();
 
+  // Peak memory used by this sorter, set immediately before each spill
+  private long peakMemoryUsage;
```

Comment:
Style nit: in the Tungsten code I prefer to use `/**` style comments even for private variables since it makes things marginally nicer in IDEs.  If we want to be really explicit, we might want to name this `peakMemoryUsageBytes`.